xc/common/utils/prjxray_import_tile_capacity.py

```diff
@@ -35,41 +46,58 @@ def main():
     parser.add_argument('--tile_type', required=True)
     parser.add_argument('--pb_types', required=True)
     parser.add_argument('--pin_assignments', required=True)
+    parser.add_argument('--sort_sites', action="store_true")
+    parser.add_argument(
+        '--unused_wires',
+        help="Comma seperated list of site wires to exclude in this tile."
+    )
 
     args = parser.parse_args()
 
     with open(args.pin_assignments) as f:
         pin_assignments = json.load(f)
 
     db = prjxray.db.Database(args.db_root, args.part)
+    grid = db.grid()
     tile_type = db.get_tile_type(args.tile_type)
 
-    sites = {}
-
     pb_types = args.pb_types.split(',')
 
     equivalent_sites_dict = dict()
+
+    gridinfo = None
+    for tile in grid.tiles():
+        if args.tile_type in tile:
+            gridinfo = grid.gridinfo_at_tilename(tile)
+
+            break
+
+    assert gridinfo
+
     for pb_type in pb_types:
         try:
             site, equivalent_sites = pb_type.split("/")
         except ValueError:
             site = pb_type
             equivalent_sites = None
 
-        sites[site] = []
-
         equivalent_sites_dict[site] = equivalent_sites.split(
             ':'
         ) if equivalent_sites else []
 
-    for site in tile_type.get_sites():
-        if site.type not in sites.keys():
-            continue
+    sites = list()
 
+    for site in tile_type.get_sites():
         site_type = db.get_site_type(site.type)
-        input_wires, output_wires = get_wires(site, site_type)
+        input_wires, output_wires = get_wires(
+            site, site_type, args.unused_wires
+        )
+
+        sites.append((site_type, site, input_wires, output_wires))
 
-        sites[site.type].append((site, input_wires, output_wires))
+    # If enabled, sort sites by name
```

Comment:
Is there ever a reason to not sort them?

Reply:
Actually yes, I bumped into an issue after applying sorting to all the capacity, I got an error for which the SLICEM and SLICEL were swapped in the sub_tile definitions, causing the wrong prefix to be assigned, given the different placement.

Thinking about it, I might have missed to sort also when generating the architecture layout, therefore having the tiles and their location prefixes not in sync.

As a long term more robust solution though, we would need to enhance VPR so that the sub_tile index is not defined by the order of the sub tile, but by an attribute that we can control.

Reply:
Fixed the tile prefix ordering in the arch-layout, so now all sites are sorted.